File: Bachelor/Semester5/Mobile_Application_Programming/server/Validator/src/main/java/validator/ExpenseValidator.java
package validator;

import model.Expense;

public class ExpenseValidator {
    public void validate(Expense expense) throws ValidationException {
        String errors = "";
        if (expense.getName().equals("")) {
            errors += "Invalid name\n";
        }
        if (expense.getSum() <= 0) {
            errors += "Invalid sum\n";
        }
        if (!errors.equals("")) {
            throw new ValidationException(errors);
        }
    }
}

File: Bachelor/Semester5/Mobile_Application_Programming/android/testmergeandroid/app/src/main/java/validator/ExpenseValidator.kt
package validator

import model.Expense

class ExpenseValidator : Validator<Expense> {
    override fun validate(e: Expense) {
        var errors = ""
        if (e.name == "")
            errors += "Invalid name\n"
        if (e.sum <= 0)
            errors += "Invalid sum\n"
        if(errors != "")
            throw ValidationException(errors)
    }
}